src/main/java/org/openrewrite/java/template/processor/PreCondition.java

```diff
@@ -0,0 +1,58 @@
+package org.openrewrite.java.template.processor;
+
+import lombok.RequiredArgsConstructor;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+
+@RequiredArgsConstructor
+public class PreCondition {
+
+    @RequiredArgsConstructor
+    public static class Rule extends PreCondition {
+        final String rule;
+
+        @Override
+        public String toString() {
+            return rule;
+        }
+    }
+
+    @RequiredArgsConstructor
+    public static class Or extends PreCondition {
+        final Set<PreCondition> rules;
+        final int indent;
+
+        @Override
+        public String toString() {
+            return joinPreconditions(rules, "or", indent);
+        }
+    }
+
+    @RequiredArgsConstructor
+    public static class And extends PreCondition {
+        final Set<PreCondition> rules;
+        final int indent;
+
+        @Override
+        public String toString() {
+            return joinPreconditions(rules, "and", indent);
+        }
+    }
+
+    private static String joinPreconditions(Collection<PreCondition> rules, String op, int indent) {
+        if (rules.isEmpty()) {
+            return "";
+        } else if (rules.size() == 1) {
+            return rules.iterator().next().toString();
+        }
+        char[] indentChars = new char[indent];
+        Arrays.fill(indentChars, ' ');
+        String indentStr = new String(indentChars);
```

Comment:
String.format also has a way to add left/right padding with `%4s` for instance; if you replace the string concatenation below then that might help readability and remove the need for these repeated strings.

Reply:
Hey, I just copied this function 😉

Reply:
I don't think it get much clearer when using things like `%4s`. I think only be much unclearer if you write it like:

```java
String.format("Preconditions.%s(\n%" + indent + "s%s" + indentStr.substring(0, indent - 4) + ')',
                op, String.join(",\n" + indentStr, preconditions));
```